pkg/registry/apis/datasource/alt/directlookup.go

```diff
@@ -0,0 +1,89 @@
+package alt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/grafana/grafana/pkg/apis/datasource/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/registry/apis/datasource"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/utils"
+)
+
+func ProvideSQLBasedPluginConfigs(sql db.DB) datasource.PluginConfigProvider {
```

Comment:
Nit. Not a big fan re-implementing existing services/stores, but fine if we want to move fast/explore things with test data I guess. 

Suggest to use/integrate with the existing datasource store, see below, instead of `db.DB` here since that that a `db.DB` as dependency?
https://github.com/grafana/grafana/blob/a8dec1916bbfc499fa18f9695feae74a6d705742/pkg/services/datasources/service/store.go#L38-L41

The optimal though would be to integrate with the existing datasource service since it already has logic for aliases etc. This might be required to be able to get a hold of the secrets since there's a crazy amount of services needed for that it seems.

Reply:
This is here to show what is necessary -- our current services/stores have sooooooo many years of abstractions and redirection built into them that it is hard to remember it is actually a very simple interface.

With the move to k8s -- most of this code will all be removed anyway, so the sooner we identify what is config vs runtime, the better